common-pg/src/main/scala/com/socrata/pg/store/PGSecondaryDatasetMapWriter.scala

```diff
@@ -148,29 +150,71 @@ class PGSecondaryDatasetMapWriter[CT](override val conn: Connection,
     }
   }
 
-  def insertLocalRollupQuery = "INSERT INTO rollup_map (name, copy_system_id, soql, table_name) VALUES (?, ?, ?, ?)"
-  def updateLocalRollupQuery = "UPDATE rollup_map SET soql = ?, table_name = ? WHERE name = ? AND copy_system_id = ?"
-  override def createOrUpdateRollup(copyInfo: CopyInfo, name: RollupName, soql: String, rawSoql: Option[String]): LocalRollupInfo = {
-    val tableName = LocalRollupInfo.tableName(copyInfo, name)
+  def insertLocalRollupQuery = "INSERT INTO rollup_map (name, copy_system_id, soql, table_name) VALUES (?, ?, ?, ?) returning system_id"
+  def updateLocalRollupQuery = "UPDATE rollup_map SET soql = ? WHERE name = ? AND copy_system_id = ? returning system_id,table_name"
+  def createOrUpdateRollupSecondary(copyInfo: CopyInfo, name: RollupName, soql: String, rawSoql: Option[String]): Option[LocalRollupInfo] = {
     rollup(copyInfo, name) match {
       case Some(_) =>
         using(conn.prepareStatement(updateLocalRollupQuery)) { stmt =>
           stmt.setString(1, soql)
-          stmt.setString(2, tableName)
-          stmt.setString(3, name.underlying)
-          stmt.setLong(4, copyInfo.systemId.underlying)
-          t("create-or-update-rollup", "action" -> "update", "copy-id" -> copyInfo.systemId, "name" -> name)(stmt.executeUpdate())
+          stmt.setString(2, name.underlying)
+          stmt.setLong(3, copyInfo.systemId.underlying)
+          t("create-or-update-rollup", "action" -> "update", "copy-id" -> copyInfo.systemId, "name" -> name)(
+            using(stmt.executeQuery()){resultSet =>
+              if(resultSet.next()){
+                return Some(new LocalRollupInfo(copyInfo, name, soql, resultSet.getString("table_name"),RollupId(resultSet.getLong("system_id"))))
+              }
+            }
+          )
         }
       case None =>
+
+        val tableName = LocalRollupInfo.tableName(copyInfo, name,getNextRollupTableNameSequence.toString)
         using(conn.prepareStatement(insertLocalRollupQuery)) { stmt =>
           stmt.setString(1, name.underlying)
           stmt.setLong(2, copyInfo.systemId.underlying)
           stmt.setString(3, soql)
           stmt.setString(4, tableName)
-          t("create-or-update-rollup", "action" -> "insert", "copy-id" -> copyInfo.systemId, "name" -> name)(stmt.executeUpdate())
+          t("create-or-update-rollup", "action" -> "insert", "copy-id" -> copyInfo.systemId, "name" -> name)(
+            using(stmt.executeQuery()){resultSet =>
+              if(resultSet.next()){
+                return Some(new LocalRollupInfo(copyInfo, name, soql, tableName,RollupId(resultSet.getLong("system_id"))))
+              }
+            }
+          )
         }
     }
-    new LocalRollupInfo(copyInfo, name, soql, tableName)
+    None
+  }
+  def createRollupRelationship = "insert into rollup_relationship_map (rollup_system_id, referenced_copy_system_id) values (?,?) on conflict do nothing"
```

Comment:
does not need to be a def

Reply:
def is how we declare the other query strings.
I agree this could be a val.
I guess the benefit of a def is that its not stored in memory?
Probably not really a concern either way.
I'd prefer to keep to similar to the rest of the code style.

Reply:
fair enough. A def allows for parameters, which you aren't using, nor are you relying on the laziness of an unevaluated fn.
up2u :)